src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1573,9 +1573,11 @@ def generate(
                     language_token = generation_config.language
                 elif generation_config.language in TO_LANGUAGE_CODE.keys():
                     language_token = f"<|{TO_LANGUAGE_CODE[generation_config.language]}|>"
+                elif generation_config.language in LANGUAGES.keys():
+                    language_token = f"<|{generation_config.language}|>"
                 else:
                     raise ValueError(
-                        f"Unsupported language: {self.language}. Language should be one of:"
+                        f"Unsupported language: {generation_config.language!r}. Language should be one of:"
                         f" {list(TO_LANGUAGE_CODE.keys()) if generation_config.language in TO_LANGUAGE_CODE.keys() else list(TO_LANGUAGE_CODE.values())}."
```

Comment:
Wonder if here we can also give a more targeted error message? E.g. list the language codes if a length-2 input, otherwise the language names? See:
https://github.com/huggingface/transformers/blob/57ffd8ab4c833e26b2288769f6031f94870a102c/src/transformers/models/whisper/tokenization_whisper.py#L405-L409